pkg/network/dns/stats.go

```diff
@@ -63,16 +63,15 @@ type stateValue struct {
 type dnsStatKeeper struct {
 	mux sync.Mutex
 	// map a DNS key to a map of domain strings to a map of query types to a map of  DNS stats
-	stats              StatsByKeyByNameByType
-	state              map[stateKey]stateValue
-	expirationPeriod   time.Duration
-	exit               chan struct{}
-	maxSize            int // maximum size of the state map
-	deleteCount        int
-	processedStats     int64
-	lastProcessedStats int64
```

Comment:
Are we overflowing here? Trying to figure out why we are changing this.

Reply:
the math doesn't make sense in the current impl. we store `processedStats` and `lastProcessedStats`. we increment the former and then when we log we set the latter to the former, and the former to 0. take this example.
start:
`processedStats`:`0`
`lastProcessedStats`:`0`
stuff happens:
`processedStats`:`50`
`lastProcessedStats`:`0`
we log and print 50 - 0 = 50, then reset the numbers:
`processedStats`:`0`
`lastProcessedStats`:`50`
more stuff happens
`processedStats`:`30`
`lastProcessedStats`:`50`
now we log again and print 30 - 50 = -20, which doesn't make sense. there's no need to keep `lastProcessedStats` at all, with this PR we just keep `processedStats` and reset it to 0 every time we log. 
the prometheus metric was always tracked separately.

Reply:
Will you please add this explanation to the PR description so future reviewers can understand this PR